### ISSUE TITLE
[1.20.1][1.19.3][1.18.2] Do not dismount players on dim change- Mod conflict with Ad Astra, Beyond earth mod and Space Dimensions mod.

### ISSUE THREAD

**Issue by ProsperCraft** [Minecraft-Transit-Railway/Minecraft-Transit-Railway#427]
**Expected Behaviour**
I expect that this would not dismount you from your rocket when changing dimentions.

**Actual Behaviour**
When you fly a rocket to the moon from either of the above mods MTR dismounts you so you fall without your rocket and it blows up when it hits the ground.

**Reproduction Steps**
Steps to reproduce the behaviour:

1.Install Beyond earth, place a rocket pad in a 3x3 multiblock
2. place a tier 1 rocket on the pad
3. shift click and add a bucket of rocket fuel
4. get in the rocket and press space
5. select earth, then moon
6. spawn without a rocket/lander

You can do the same with Space dimentions mod which is an entirely different mod but the same thing happens.

**Environment**

- Minecraft Version: 1.18.2 and including all versions up to 1.20.1
- Installed Mods: MTR-forge-1.18.2-3.0.1.jar  Beyond-Earth-1.18.2-6.2.jar  architectury-4.5.76-forge.jar

**Additional Information**
If you could make it so player is not dismounted on dim change that would be great.

**Comment by Dzuco** [comment]
I have the same problem, is there any news on this?

**Comment by kettle-7** [comment]
Not currently, though I'm sure if there is news you'll find it here.

**Comment by ProsperCraft** [comment]
This is still a problem in 1.19.3 with Ad Astra mod.

**Comment by ThatGravyBoat** [comment]
The reason this happens is because of this line of code here: https://github.com/jonafanho/Minecraft-Transit-Railway/blob/master/common/src/main/java/mtr/data/RailwayDataCoolDownModule.java#L64
I used this mixin here to see where dismount was getting called
```java
@Mixin(Entity.class)
public class FindTheCode {

    @Shadow @Nullable private Entity vehicle;

    @Inject(method = "removeVehicle", at = @At("HEAD"))
    private void onDismount(CallbackInfo ci) {
        if (this.vehicle instanceof Lander) {
            Thread.dumpStack();
        }
    }
}
```

and got a stack trace leading back to that line of code.

**Comment by jonafanho** [comment]
Thanks for investigating! I think I shouldn't call dismount when the player first switches dimensions.

**Comment by Edurd121** [comment]
Do we know if this will be resolved in the near future?

**Comment by kettle-7** [comment]
> Thanks for investigating! I think I shouldn't call dismount when the player first switches dimensions.

Would this mean that we could take a train to the nether for example ?

**Comment by Dragonfly-Totem** [comment]
For what it's worth, this issue _also_ is present in the Fabric version of the mod in MC 1.19.3.

**Comment by Kojimo** [comment]
Also on 1.19.4
Is very gamebreaking... any suggestion to get over it?

**Comment by ProsperCraft** [comment]
From what I understand it has been fixed, just not included in the releases.

**Comment by Kojimo** [comment]
> From what I understand it has been fixed, just not included in the releases.

Mmmh... doesn't seem the case to me, I have the exactly same behaviour described in the issue... can be another mod but I don't think so.
I could try today "for science" anyway with removing MTR and retry.

Anyway in the maintime i found (and tested) a strategy to temporarily overcome the problem: when you spawn in the new planet launch the command
`/ride <playerusername> mount @e[type=ad_astra:lander, limit=1, sort=nearest]`

You or someone else online on the server needs to be op anyway... I don't know even if the op player needs to be in the sameplanet as target player honestly

**Comment by ProsperCraft** [comment]
Confirming the issue still exists in 1.20.1 please fix, having your hard earned rocket fall to the planet and explode then getting killed on impact on dim change is very sad.

**Comment by AoSankaku** [comment]
Still same on 1.19.2 with Ad Astra

**Comment by BichoMESP** [comment]
Solution until fixed in the mod
[https://github.com/BichoMESP/bug-mtr](url)

**Comment by BichoMESP** [comment]
Solution until fixed in the mod.

[https://github.com/BichoMESP/bug-mtr](url)

**Comment by BichoMESP** [comment]
Solution until fixed in the mod.

[https://github.com/BichoMESP/bug-mtr](url)

**Comment by Kenny-Hui** [comment]
For an update:
This has been fixed by Ad-Astra since the beta release 1.15.7 and above (You may not want to use beta version until it's released however)
https://modrinth.com/mod/ad-astra/version/1.15.7

In addition, a new MTR Version (MTR 4) is currently in development, and the affected code no longer exists.
A quick test seems to suggest that this issue has been resolved even on older Ad-Astra version with the in-development MTR version.
(TL;DR Both sides should have resolved the compatibility respectively, but may take a while before these fixed version reaches normal players)
